packages/contract/lib/utils/index.js

```diff
@@ -272,9 +272,12 @@ const Utils = {
     { networks, currentProvider, setNetwork, network_id },
     gasLimit
   ) {
+    if (!networks && Object.keys(networks).length === 0) {
```

Comment:
This is one of the reasons I like `Map`'s over `Object`s.  It keeps track of its size so no need to count!